firmware/arduino/chessboard/uart.ino

```diff
@@ -79,46 +82,53 @@ bool validateMessageFromPi(volatile char * message)
     {
         if (isInvalidCoord(message[1]) || isInvalidCoord(message[2]))
         {
-            errorCode = INVALID_LOCATION;
+            extraByte = INVALID_LOCATION;
             currentState = ERROR;
             return false;
         }
     }
     else if (message[0] == INSTRUCTION)
     {
-        // Check if message[5] holds a valid instruction type
-        if (message[5] < ALIGN_AXIS || message[5] > RETRANSMIT)
+        // Check if message[5] holds an invalid instruction type
+        if ((message[1] != ALIGN_AXIS  || message[2] < '1' || message[2] > '3') && 
+            (message[1] != SET_ELECTROMAGNET || message[2] < '0' || message[2] > '1') &&
+             message[1] != RETRANSMIT)
         {
-            errorCode = INVALID_LOCATION;
+            extraByte = INVALID_LOCATION;
             currentState = ERROR;
             return false;
         }
+
+        // Send opcode back in the extraByte field
+        extraByte = message[1];
     }
     else
     {
         // Invalid opcode
-        errorCode = INVALID_OP;
+        extraByte = INVALID_OP;
         currentState = ERROR;
         return false;
     }
 
     // Update last valid move count
     moveCount = message[5];
 
-    errorCode = NO_ERROR;
     return true;
 }
 
 bool makeMove(volatile char * message)
 {
+    // If no error occurs to change the extraByte, it should indicate NO_ERROR
+    extraByte = NO_ERROR;
```

Comment:
As above, if no error, this should store the opcode value. only assign something that is not an opcode here if we encounter an error.